anvil/src/shell/mod.rs

```diff
@@ -80,14 +86,24 @@ impl FullscreenSurface {
     }
 }
 
-impl<BackendData: Backend> BufferHandler for AnvilState<BackendData> {
-    fn buffer_destroyed(&mut self, _buffer: &WlBuffer) {}
-}
-
 impl<BackendData: Backend> CompositorHandler for AnvilState<BackendData> {
     fn compositor_state(&mut self) -> &mut CompositorState {
         &mut self.compositor_state
     }
+    fn client_compositor_state<'a>(&self, client: &'a Client) -> &'a CompositorClientState {
+        #[cfg(feature = "xwayland")]
+        if let Some(state) = client.get_data::<XWaylandClientData>() {
+            state
+                .user_data()
+                .insert_if_missing(CompositorClientState::default);
+            return state.user_data().get::<CompositorClientState>().unwrap();
```

Comment:
Could that be directly added to `XWaylandClientData`? `launch` initializes `XWaylandClientData` where it would be possible to directly add `CompositorClientState` as a field instead of using `user_data`.